### PR TITLE
Add e2e test

### PR DESCRIPTION
**Pull request by snakazawa**
I added E2E test using Nightwatch. You can run E2E test by `npm run e2e`.
(Depended packages are updated, please execute `npm i` before that.)

Nightwatch uses Chrome as default.
If you watch on Firefox, execute `npm run e2e -- --env firefox`.

E2E test checks only client behavior.
If you want to check what is playing, execute `E2E_WAIT_TIME=10000 npm run e2e` and **check with your ears**.
If you want to run E2E test fast, execute `E2E_WAIT_TIME=1000 npm run e2e`.
(E2E_WAIT_TIME is 5000ms as default.)

### Known problems:
- **E2E test process never end** even though all tests is passed.

### Fixed bugs:
- Stop playing music when playlist is cleared.
- Update playing index to 0 when playlist is cleared.

### Not checked features in E2E test:
- Shuffle mode
- Volume change
- Mute
- Other complex conditions

Sample music URLs were chosen from *swk-ikoi#music* (and my taste).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -58,4 +58,10 @@ typings/
...
 
 /store/*
-!/store/.gitkeep
\ No newline at end of file
+!/store/.gitkeep
+
+# e2e test
+/bin/selenium-server-standalone-3.4.0.jar
+/bin/geckodriver
+/bin/chromedriver
```
Comment (unblee) [inline .gitignore]: Is this '/bin/*' part necessary?
I thought it was not necessary in my environment.

Comment (snakazawa) [inline .gitignore]: Sorry, I removed these line.

**Comment by snakazawa** [comment]
I resolved conflicts and updated `yarn.lock`.

**Comment by snakazawa** [comment]
I added 'chrome-headless' mode.
You can run headless test by `yarn run e2e -- -e chrome-headless`.

I cannot find how to Firefox headless using Nightwatch.

**Comment by unblee** [comment]
@snakazawa Please fix the conflict 🙏

**Comment by snakazawa** [comment]
I solved conflict :)

**Comment by unblee** [comment]
Umm...
After executing the test command, the following log was output.
```
$ yarn run e2e
yarn run v0.28.4
$ nightwatch --config nightwatch.conf.js
Starting selenium server... started - PID:  10249

[Index Test] Test Suite
===========================

Running:  basicTest
 ✔ Element <body> was visible after 74 milliseconds.
 ✖ Testing if element <body> contains text: "jukebox".  - expected "jukebox" but got: "  volume_up
arrow_forward
shuffle
delete_sweep
send"
    at Object.basicTest (/home/unblee/workspace/src/github.com/unblee/jukebox/e2e/tests/index_test.js:32:64)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)


FAILED:  1 assertions failed and 1 passed (3.973s)

 _________________________________________________

 TEST FAILURE:  1 assertions failed, 1 passed. (4.354s)

 ✖ index_test

   - basicTest (3.973s)
   Testing if element <body> contains text: "jukebox".  - expected "jukebox" but got: "  volume_up
   arrow_forward
   shuffle
   delete_sweep
   send"
       at Object.basicTest (/home/unblee/workspace/src/github.com/unblee/jukebox/e2e/tests/index_test.js:32:64)
       at _combinedTickCallback (internal/process/next_tick.js:131:7)
       at process._tickCallback (internal/process/next_tick.js:180:9)
   SKIPPED:
   - oneLoopTest
   - playlistLoopTest

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by snakazawa** [comment]
That error was caused by replacing title text with logo.
I fixed the error, and checked E2E test is all green !

**Comment by unblee** [comment]
Oh... 😨 
A normal test command succeeds.
But after executing the headless test command(`yarn run e2e -- -e chrome-headless`), the following log was output.
```
$ yarn run e2e -- -e chrome-headless
yarn run v0.28.4
$ nightwatch --config nightwatch.conf.js "-e" "chrome-headless"
Starting selenium server... started - PID:  11744

[Index Test] Test Suite
===========================

Running:  basicTest
 ✔ Element <body> was visible after 61 milliseconds.
 ✔ Element <a.playlist-content:nth-child(5)> was present after 6030 milliseconds.
 ✔ Testing if element <a[title=Play]> is present.
 ✔ Testing if length of playlist is 5
 ✔ Element <a[title=Pause]> was present after 29 milliseconds.
 ✔ Testing if element <a[title=Pause]> is present.
 ✔ Testing if length of playlist is 5
 ✔ Testing if current track number is 1
 ✔ Element <a.playlist-content:nth-child(5)> was not present after 544 milliseconds.
 ✔ Testing if element <a[title=Pause]> is present.
 ✔ Testing if length of playlist is 4
 ✔ Testing if current track number is 1
 ✔ Testing if element <a[title=Prev]> has css class: "deactivate".
 ✔ Element <a[title=Play]> was present after 19 milliseconds.
 ✔ Testing if element <a[title=Play]> is present.
 ✔ Testing if length of playlist is 4
 ✔ Testing if current track number is 1
 ✔ Element <a[title=Pause]> was present after 546 milliseconds.
 ✔ Testing if element <a[title=Pause]> is present.
 ✔ Testing if length of playlist is 4
 ✔ Testing if current track number is 3
 ✖ Timed out while waiting for element <a[title=Play]> to be present for 5000 milliseconds.  - expected "found" but got: "not found"
    at Object.basicTest (/home/unblee/workspace/src/github.com/unblee/jukebox/e2e/tests/index_test.js:91:8)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)


FAILED:  1 assertions failed and 21 passed (42.293s)

 _________________________________________________

 TEST FAILURE:  1 assertions failed, 21 passed. (42.667s)

 ✖ index_test

   - basicTest (42.293s)
   Timed out while waiting for element <a[title=Play]> to be present for 5000 milliseconds.  - expected "found" but got: "not found"
       at Object.basicTest (/home/unblee/workspace/src/github.com/unblee/jukebox/e2e/tests/index_test.js:91:8)
       at _combinedTickCallback (internal/process/next_tick.js:131:7)
       at process._tickCallback (internal/process/next_tick.js:180:9)
   SKIPPED:
   - oneLoopTest
   - playlistLoopTest
```

**Comment by snakazawa** [comment]
kibiC

**Comment by unblee** [comment]
👍

**Comment by snakazawa** [comment]
I have no idea of the cause of the bug......

**Comment by unblee** [comment]
Ummmm...

**Comment by snakazawa** [comment]
I solved the problem!!!

That cause is window size.
Default window size of chrome headless mode is small, so I added `window-size` args in `nightwatch.conf.js`.

**Comment by snakazawa** [comment]
I checked that tests is all green in chrome, chrome-headless and firefox environments after merging with current master.

**Comment by unblee** [comment]
All OK!!! 💯 💯 
Thank you so, so much!!!!! 🙏 🙏 🙏

**Pull request merged**